src/dist/mod.rs

```diff
@@ -30,6 +32,124 @@ pub mod http;
 #[cfg(test)]
 mod test;
 
+// TODO: paths (particularly outputs, which are accessed by an unsandboxed program)
+// should be some pre-sanitised AbsPath type
```

Comment:
Maybe as a first pass you could just have a newtype around `PathBuf` that implements `AsRef`?